src/gil.rs

```diff
@@ -128,18 +129,35 @@ where
     ffi::Py_InitializeEx(0);
 
     // Safety: the GIL is already held because of the Py_IntializeEx call.
-    #[allow(deprecated)] // TODO: remove this with the GIL Refs feature in 0.22
-    let pool = GILPool::new();
+    #[cfg(feature = "gil-refs")]
+    #[allow(deprecated)]
+    let result = {
+        let pool = GILPool::new();
+
+        // Import the threading module - this ensures that it will associate this thread as the "main"
+        // thread, which is important to avoid an `AssertionError` at finalization.
+        pool.python().import_bound("threading").unwrap();
 
-    // Import the threading module - this ensures that it will associate this thread as the "main"
-    // thread, which is important to avoid an `AssertionError` at finalization.
-    pool.python().import_bound("threading").unwrap();
+        // Execute the closure.
+        let result = f(pool.python());
 
-    // Execute the closure.
-    let result = f(pool.python());
+        // Drop the pool before finalizing.
+        drop(pool);
 
-    // Drop the pool before finalizing.
-    drop(pool);
+        result
+    };
+
+    #[cfg(not(feature = "gil-refs"))]
+    let result = {
+        let _guard = GILGuard::acquire_unchecked();
```

Comment:
Ah, exactly, this is calling `PyGILState_Ensure` for which there is no reason to in the entry points.